tests/cpp_validation_tests/test_validation.cpp

```diff
@@ -531,21 +644,31 @@ void execute_test(CaseParam const& param, T&& func) {
 
 void validate_single_case(CaseParam const& param) {
     execute_test(param, [&]() {
-        auto const validation_case = create_validation_case(param);
         auto const output_prefix = get_output_type(param.calculation_type, param.sym);
-        auto const result = create_result_dataset(validation_case.input, output_prefix);
-
-        // create model and run
-        MainModel model{50.0, validation_case.input.const_dataset, 0};
-        CalculationFunc const func = calculation_func(param);
-
-        ConstDataset empty{false, 1, "update", meta_data_gen::meta_data};
-        func(model, calculation_method_mapping.at(param.calculation_method), result.dataset, empty, -1);
-        assert_result(result.const_dataset, validation_case.output.value().const_dataset, param.atol, param.rtol);
+        auto const validation_case = create_validation_case(param, output_prefix);
+        auto const result = create_result_dataset(validation_case.output.value(), output_prefix);
+
+        // create and set options
+        power_grid_model_cpp::Options options{};
+        options.set_calculation_type(calculation_type_mapping.at(param.calculation_type));
+        options.set_calculation_method(calculation_method_mapping.at(param.calculation_method));
+        options.set_symmetric(param.sym ? 1 : 0);
+        options.set_err_tol(1e-8);
+        options.set_max_iter(20);
+        options.set_threading(-1);
+        options.set_short_circuit_voltage_scaling(sc_voltage_scaling_mapping.at(param.short_circuit_voltage_scaling));
+        options.set_tap_changing_strategy(optimizer_strategy_mapping.at(param.tap_changing_strategy));
```

Comment:
I added a separate `get_options` function as suggested. However, the magic numbers still remain (and were there originally) since the `CaseParam`does not contain them because they can not be extracted from the data files (they are just not present there).